### DISCUSSION TITLE
Porting to esp32

### DISCUSSION THREAD

**Discussion by hell1337hound** [HASwitchPlate/HASPone#172] [Ideas]
Hello, love the project but i was wondering how hard it would be to move to the ESP32? I've got a 7" display and it lags a bit on the esp8266

**Comment by aderusha** [comment]
While it's technically possible, the physical side of the project wouldn't work with any ESP32 boards that I am currently aware of due to available PCB space.  The ESP8266 is doing very little work here, it's simply taking Nextion commands from MQTT and sending them to the display.  If performance isn't great, it's likely on the display itself and re-platforming to ESP32 isn't going to make much of an impact either way.

I've moved this over to the Discussion area as it isn't really an issue per se, however it might be useful for discussion!

**Comment by BenJewell** [comment]
I just found out about this project and came here wondering the same thing. I would like the ESP32 for the 5ghz Wi-Fi support, lots of congestion here.

**Reply by fvanroie** [reply]
> I would like the ESP32 for the 5ghz Wi-Fi support

ESP32 doesn't support 5 GHz.

**Reply by stampeder** [reply]
Well, not quite. The ESP32-C5 supports 5 GHZ wifi and also ZIGBEE....

**Reply by fvanroie** [reply]
The Espressif ESP32-C5 was announced in 2022 but entered mass production in April 2025.

**Reply by stampeder** [reply]
Yup. I have the dev kit ESP32-C5-DevKitC-1 that I am about to start playing with. My interest is in the Zigbee capabilities primarily. 5G will come later. Why not order one and test it out? They are cheap.....
Right now too busy getting 3 HASPones going. Good luck.
